hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/keyvalue/TestKeyValueHandlerWithUnhealthyContainer.java

```diff
@@ -249,6 +262,64 @@ public void testMarkContainerUnhealthyInFailedVolume() throws IOException {
     verify(mockIcrSender, atMostOnce()).send(any());
   }
 
+  @Test
+  public void testDispatcherTriggersOnDemandScanForUnhealthyContainer() throws IOException {
```

Comment:
Is there a simpler way to test this with `TestOnDemandContainerScannerIntegration`? For example:
- Change permissions on the volume of the one DN in the cluster so all writes fail.
- Issue a write request with `writeDataToOpenContainer`
- Verify the container is marked UNHEALTHY
- Check `OnDemandScannerMetrics` and `ContainerData#getDataScanTimestamp` to verify an on-demand scan was triggered.